### ISSUE TITLE
Autopilot integration is missing default fields

### ISSUE THREAD

**Issue by jhealey5** [verbb/formie#259]
There's a bunch of default fields on Autopilot, mentioned in their API docs (https://autopilot.docs.apiary.io/#reference/api-methods/addupdate-contact/add-or-update-contact), that are missing from the plugin.

I've taken the time to add them so I thought I'd add them here so they can be updated easily

https://github.com/verbb/formie/blob/craft-3/src/integrations/emailmarketing/Autopilot.php#L74-L112

```
$listFields = array_merge([
    new IntegrationField([
        'handle' => 'Email',
        'name' => Craft::t('formie', 'Email'),
        'required' => true,
    ]),
    new IntegrationField([
        'handle' => 'Title',
        'name' => Craft::t('formie', 'Title'),
    ]),
    new IntegrationField([
        'handle' => 'FirstName',
        'name' => Craft::t('formie', 'First Name'),
    ]),
    new IntegrationField([
        'handle' => 'LastName',
        'name' => Craft::t('formie', 'Last Name'),
    ]),
    new IntegrationField([
        'handle' => 'Salutation',
        'name' => Craft::t('formie', 'Salutation'),
    ]),
    new IntegrationField([
        'handle' => 'Company',
        'name' => Craft::t('formie', 'Company'),
    ]),
    new IntegrationField([
        'handle' => 'NumberOfEmployees',
        'name' => Craft::t('formie', 'Number Of Employees'),
    ]),
    new IntegrationField([
        'handle' => 'Industry',
        'name' => Craft::t('formie', 'Industry'),
    ]),
    new IntegrationField([
        'handle' => 'Phone',
        'name' => Craft::t('formie', 'Phone'),
    ]),
    new IntegrationField([
        'handle' => 'MobilePhone',
        'name' => Craft::t('formie', 'Mobile Phone'),
    ]),
    new IntegrationField([
        'handle' => 'Fax',
        'name' => Craft::t('formie', 'Fax'),
    ]),
    new IntegrationField([
        'handle' => 'Website',
        'name' => Craft::t('formie', 'Website'),
    ]),
    new IntegrationField([
        'handle' => 'MailingStreet',
        'name' => Craft::t('formie', 'Mailing Street'),
    ]),
    new IntegrationField([
        'handle' => 'MailingCity',
        'name' => Craft::t('formie', 'Mailing City'),
    ]),
    new IntegrationField([
        'handle' => 'MailingState',
        'name' => Craft::t('formie', 'Mailing State'),
    ]),
    new IntegrationField([
        'handle' => 'MailingPostalCode',
        'name' => Craft::t('formie', 'Mailing Postal Code'),
    ]),
    new IntegrationField([
        'handle' => 'MailingCountry',
        'name' => Craft::t('formie', 'Mailing Postal Country'),
    ]),
    new IntegrationField([
        'handle' => 'owner_name',
        'name' => Craft::t('formie', 'Owner Name'),
    ]),
    new IntegrationField([
        'handle' => 'LeadSource',
        'name' => Craft::t('formie', 'Lead Source'),
    ]),
    new IntegrationField([
        'handle' => 'Status',
        'name' => Craft::t('formie', 'Status'),
    ]),
    new IntegrationField([
        'handle' => 'Twitter',
        'name' => Craft::t('formie', 'Twitter'),
    ]),
    new IntegrationField([
        'handle' => 'LinkedIn',
        'name' => Craft::t('formie', 'LinkedIn'),
    ]),
    new IntegrationField([
        'handle' => 'unsubscribed',
        'name' => Craft::t('formie', 'Unsubscribed'),
    ]),
], $this->_getCustomFields($fields));
````

and https://github.com/verbb/formie/blob/craft-3/src/integrations/emailmarketing/Autopilot.php#L139-L164

```
// Pull out email, as it needs to be top level
$email             = ArrayHelper::remove($fieldValues, 'Email');
$firstName         = ArrayHelper::remove($fieldValues, 'FirstName');
$lastName          = ArrayHelper::remove($fieldValues, 'LastName');
$company           = ArrayHelper::remove($fieldValues, 'Company');
$phone             = ArrayHelper::remove($fieldValues, 'Phone');
$mobilePhone       = ArrayHelper::remove($fieldValues, 'MobilePhone');
$website           = ArrayHelper::remove($fieldValues, 'Website');
$leadSource        = ArrayHelper::remove($fieldValues, 'LeadSource');
$status            = ArrayHelper::remove($fieldValues, 'Status');
$title             = ArrayHelper::remove($fieldValues, 'Title');
$salutation        = ArrayHelper::remove($fieldValues, 'Salutation');
$numberOfEmployees = ArrayHelper::remove($fieldValues, 'NumberOfEmployees');
$industry          = ArrayHelper::remove($fieldValues, 'Industry');
$fax               = ArrayHelper::remove($fieldValues, 'Fax');
$mailingStreet     = ArrayHelper::remove($fieldValues, 'MailingStreet');
$mailingCity       = ArrayHelper::remove($fieldValues, 'MailingCity');
$mailingState      = ArrayHelper::remove($fieldValues, 'MailingState');
$mailingPostalCode = ArrayHelper::remove($fieldValues, 'MailingPostalCode');
$mailingCountry    = ArrayHelper::remove($fieldValues, 'MailingCountry');
$owner_name        = ArrayHelper::remove($fieldValues, 'owner_name');
$twitter           = ArrayHelper::remove($fieldValues, 'Twitter');
$linkedIn          = ArrayHelper::remove($fieldValues, 'LinkedIn');
$unsubscribed      = ArrayHelper::remove($fieldValues, 'unsubscribed');

$payload = [
    'contact' => [
        'Email'             => $email,
        'FirstName'         => $firstName,
        'LastName'          => $lastName,
        'Company'           => $company,
        'Phone'             => $phone,
        'MobilePhone'       => $mobilePhone,
        'Website'           => $website,
        'LeadSource'        => $leadSource,
        'Status'            => $status,
        'MailingCity'       => $mailingCity,
        'Title'             => $title,
        'Salutation'        => $salutation,
        'NumberOfEmployees' => $numberOfEmployees,
        'Industry'          => $industry,
        'Fax'               => $fax,
        'MailingStreet'     => $mailingStreet,
        'MailingCity'       => $mailingCity,
        'MailingState'      => $mailingState,
        'MailingPostalCode' => $mailingPostalCode,
        'MailingCountry'    => $mailingCountry,
        'owner_name'        => $owner_name,
        'Twitter'           => $twitter,
        'LinkedIn'          => $linkedIn,
        'unsubscribed'      => $unsubscribed,
        '_autopilot_list'   => $this->listId,
        'custom'            => $fieldValues,
    ]
];
```

Hope that helps!

**Comment by engram-design** [comment]
Thanks for this, looks like I didn't add all the custom fields at the time, so appreciate the contribution!

As for the `_autopilot_list` I could've sworn that was working correctly at some point, but thanks again.

**Comment by engram-design** [comment]
Should be fixed for the next release. To get the fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.3.3",
  "...": "..."
}
```

Then run `composer update`.

**Comment by engram-design** [comment]
Fixed in [1.3.4](https://github.com/verbb/formie/releases/tag/1.3.4)
